src/lua/widget/container.c

```diff
@@ -154,16 +154,38 @@ static int container_numindex(lua_State*L)
       GtkWidget *searched_widget = g_list_nth_data(children,index);
       gtk_container_remove(GTK_CONTAINER(container->widget),searched_widget);
     } else {
-      luaL_error(L,"Incorrect index or value when setting the child of a container : you can only append, not change a child in the middle");
+      GList *current = g_list_nth(children, index);
+      gtk_container_remove(GTK_CONTAINER(container->widget), current->data);
+      for(current = current->next; current != NULL; current = current->next)
+      {
+        g_object_ref(current->data);
+        gtk_container_remove(GTK_CONTAINER(container->widget), current->data);
+      }
+
+      lua_widget widget;
+      luaA_to(L, lua_widget, &widget, 3);
+      gtk_container_add(GTK_CONTAINER(container->widget), widget->widget);
+
+      for(current = g_list_nth(children, index + 1); current != NULL; current = current->next)
+      {
+        gtk_container_add(GTK_CONTAINER(container->widget), current->data);
+        g_object_unref(current->data);
+      }
     }
     g_list_free(children);
     return 0;
   } else {
-    GtkWidget *searched_widget = g_list_nth_data(children,index);
+    if(index < 0 || index >= g_list_length(children))
+    {
+      lua_pushnil(L);
+    }
+    else
+    {
+      GtkWidget *searched_widget = g_list_nth_data(children, index);
+      lua_getuservalue(L, 1);
+      luaA_push(L, lua_widget, &searched_widget);
+    }
     g_list_free(children);
-    lua_getuservalue(L,1);
-    lua_pushlightuserdata(L,searched_widget);
-    lua_gettable(L,-2);
```

Comment:
That last line has disapeared and that looks weird to me

the old code returned container->uservalue[searched_widget]  your new code returns searched_widget directely.

Could you check what's going on here ? IIRC because of the way GTK and lua interact, you need to go throught that gymnastic or you risk returning the gtk-widget instead of the lua widget...